mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutorStarter.java

```diff
@@ -68,6 +69,7 @@ protected void shutDown() throws Exception {
             .exceptionally(throwable -> null)
             .thenCompose(dontCare -> Services.stopAsync(highAvailabilityServices,
                 MoreExecutors.directExecutor()))
+            .thenRunAsync(rpcSystem::close)
```

Comment:
Should we use the caller thread, in this case, using MoreExecutors.directExecutor()?